service/extensions/extensions.go

```diff
@@ -92,7 +93,7 @@ func (bes *Extensions) Shutdown(ctx context.Context) error {
 func (bes *Extensions) NotifyPipelineReady() error {
 	for _, extID := range bes.extensionIDs {
 		ext := bes.extMap[extID]
-		if pw, ok := ext.(extension.PipelineWatcher); ok {
+		if pw, ok := ext.(extensioncapabilities.PipelineWatcher); ok {
```

Comment:
We will need to continue to check the deprecated interface here during the deprecation process.

Reply:
Since the interface is exactly the same, I don't think that's needed: if something fulfills the old interface, it should also fulfill the new interface. I wrote an example here: https://go.dev/play/p/cAdIQq2r7Vl

Reply:
Right, of course, I got a bit mixed up there. This is completely okay then, thanks for going so far as to write an example.